lib/mr_test_lib.pm

```diff
@@ -54,6 +54,18 @@ sub load_mr_tests {
         loadtest_mr_test("$script", name => $i . '_saptune_' . $note_solution . $test, run_args => $tinfo);
         $i++;
     }
+
+    # Load 'ssh_interactive_end.pm' here instead of adding to 'schedule/sles4sap/sles4sap_gnome_saptune.yaml'
+    # otherwise 'ssh_interactive_end.pm' will be started to run before 'load_mr_tests' finished.
+    # Paste the messages which are cut out from openQA log file FYI for a better understanding:
+    #   [debug] ||| starting mr_test tests/sles4sap/saptune/mr_test.pm
+    #   [debug] ||| finished mr_test sles4sap/saptune
+    #   [debug] scheduling 1_saptune_notes .../mr_test_run.pm
+    #   [debug] ||| starting ssh_interactive_end tests/publiccloud/ssh_interactive_end.pm
+    #   [debug] ||| finished ssh_interactive_end publiccloud
+    #   [debug] ||| starting 1_saptune_notes .../mr_test_run.pm
+    #   [debug] ||| finished 1_saptune_notes lib
+    loadtest_mr_test('tests/publiccloud/ssh_interactive_end', run_args => $args) if get_var('PUBLIC_CLOUD_SLES4SAP');
```

Comment:
be aware that `ssh_interactive_end` has test flags `fatal` and `publiccloud_multi_module` ( see  https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/publiccloud/ssh_interactive_end.pm#L23 )  . which means that cleanup will happen only in case of module failure  see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/basetest.pm#L120 . I suspect that this is not what you expecting

Reply:
> be aware that `ssh_interactive_end` has test flags `fatal` and `publiccloud_multi_module` ( see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/publiccloud/ssh_interactive_end.pm#L23 ) . which means that cleanup will happen only in case of module failure see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/basetest.pm#L120 . I suspect that this is not what you expecting

When all test module passed, cleanup is done: https://openqaworker15.qa.suse.cz/tests/65672#step/ssh_interactive_end/28
When one of the test module failed, cleanup is done also: https://openqaworker15.qa.suse.cz/tests/67256#step/ssh_interactive_end/28
https://openqaworker15.qa.suse.cz/tests/55628#step/ssh_interactive_end/4

Reply:
ok I didn't realize that `ssh_interactive_end` is not ancestor of basetest. You are correct , ignore my message